Extension/src/LanguageServer/client.ts

```diff
@@ -2401,6 +2401,12 @@ export class DefaultClient implements Client {
             if (modifiedConfig.compileCommands) {
                 modifiedConfig.compileCommands = cppProperties.resolvePath(modifiedConfig.compileCommands, os.platform() === "win32");
             }
+
+            if (modifiedConfig.includePath) {
```

Comment:
I think `updateServerOnFolderSettingsChange` is the right place for this fix (and to resolve all paths).  That function calls `updateConfigurationPathsArray` and a bunch of similar functions on each of the settings we process.  Those functions call `resolveAndSplit` on paths.  If you resolve relative to absolute paths there, it will apply to all settings paths that call it.

Reply:
I don't think this is the right place to fix up these paths.  This is literally right before the info gets sent to the server, so none of the changes made here apply to use of those values on the TypeScript side.  I had added this bit regarding `modifiedConfig` to make some adjustments specifically for `compilerArgsLegacy`.  The right place to be fixing up strings to ensure variables are fully resolved, is in `updateServerOnFolderSettingsChange`.  That is where configuration paths are processed, multiple levels of defaults applied, variables expanded, etc..